src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -176,6 +176,11 @@ public ODataMessageQuotas MessageQuotas
         /// </summary>
         public bool ReadUntypedAsString { get; set; }
 
+        /// <summary>
+        /// Gets or sets current request Uri.
+        /// </summary>
+        public Uri RequestUri { get; set; }

```

Comment:
What's the relationship between the new RequestUri property and the existing BaseUri property?  When does RequestUri get set? Can we just use BaseUri?

Reply:
BaseUri is for Atom format and may be deprecated in the future. See [https://github.com/OData/odata.net/blob/9d625ffdece85ea51c7e0823029f4597bc288624/src/Microsoft.OData.Core/ODataMessageReaderSettings.cs#L98](https://github.com/OData/odata.net/blob/9d625ffdece85ea51c7e0823029f4597bc288624/src/Microsoft.OData.Core/ODataMessageReaderSettings.cs#L98).

---
In reply to: [203896122](https://github.com/OData/odata.net/pull/1211#discussion_r203896122) [](ancestors = 203896122)

Reply:
The reason it was to be deprecated was because it was only used in Atom. The reason it was only used in Atom was because we previously didn't support relative URLs in JSON.  Now that we support relative URLs in JSON, it seems like we should use this (existing) property, rather than deprecate it and add a new one for the same purpose.

---
In reply to: [204170332](https://github.com/OData/odata.net/pull/1211#discussion_r204170332) [](ancestors = 204170332,203896122)

Reply:
I tried using BaseUrl but it fails in multiple cases when the baseurl is the $metadata url, which is common, among other failing cases. 

---
In reply to: [205241623](https://github.com/OData/odata.net/pull/1211#discussion_r205241623) [](ancestors = 205241623,204170332,203896122)